src/renderers/webgl/WebGLProgram.js

```diff
@@ -418,6 +418,8 @@ function WebGLProgram( renderer, cacheKey, parameters, bindingStates ) {
 
 		prefixVertex = [
 
+			'#define SHADER_NAME ' + parameters.shaderName,
```

Comment:
@Mugen87 Any reason why adding this to `RawShaderMaterial` would be ill-advised?

Reply:
This seems to break raw shaders that started with `#version 300 es` (since the version directive must be the first thing in the shader source). Any advice on how to work around this? This is the same issue reported by @orme below.

Example: https://github.com/foxglove/three-text/blob/45657e532f03bc97a23e7f40b4ee603598e842da/src/LabelPool.ts#L14